### PR TITLE
ISPN-11444 Upgrade JUnit version 5

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-11444

### CODE REVIEW

**Comment by karesti** [comment]
@danberindei do you think anything else is needed at this point ?

File: pom.xml [hunk 1]
```diff
@@ -5081,6 +5102,7 @@
             <plugins>
                <plugin>
                   <artifactId>maven-surefire-plugin</artifactId>
+                  <version>2.22.2</version>
```
Comment (danberindei) [inline pom.xml:5105]: Why not `${maven.version.surefire}`?

**Comment by danberindei** [comment]
@karesti I think even after adding the dependency, the tests still run with JUnit4, because of this:

https://github.com/infinispan/infinispan/blob/38d42d23216ebaa169db885356d2a2584c33439c/commons/all/pom.xml#L192

I didn't read the [JUnit5 maven documentation](https://junit.org/junit5/docs/current/user-guide/#running-tests-build-maven) properly, it doesn't explain how to use it with Maven when the module also has a TestNG dependency. There is however an example in the Maven Surefire documentation:

https://maven.apache.org/surefire/maven-surefire-plugin/examples/junit-platform.html

Something else I think is missing is a way to check that the tests really are running with JUnit 5 and not with JUnit 4.

**Comment by karesti** [comment]
@danberindei looking tonight

**Comment by karesti** [comment]
@danberindei wdyt now ?

**Comment by danberindei** [comment]
@karesti how can we check if the tests really are running with JUnit 5?

**Comment by karesti** [comment]
@danberindei I'm trying to run junit 4 and 5 tests at the same time in another PR. It does not work (yet)

**Comment by karesti** [comment]
@danberindei I give it up. I can't make work on command line a junit 5 test

**Comment by danberindei** [comment]
@karesti I have made the JUnit 5 test driver run the tests with JUnit 5 here: https://github.com/infinispan/infinispan/commit/8dde3ef4e02f5a032f5168d639f4f3345dca16e5

Similar changes would be needed in all the modules currently using JUnit 4 to upgrade them to JUnit 5. Careful review would also be needed to make sure that the modules actually run with JUnit 5 (by running Maven with `-X` and inspecting the output), that the test count is the same, that test parameters are also preserved in the test results to keep the tests unique, and that tests that run twice in different modules (like hibernate v51 and v53) get different names in the results.

**Comment by karesti** [comment]
@danberindei the best will be to treat it all in the same PR

**Pull request closed without merging**